### DISCUSSION TITLE
How to get node embeddings per batch entry

### DISCUSSION THREAD

**Discussion by afonso-sousa** [pyg-team/pytorch_geometric#4171] [Q&A] [answered]
Hi. I am trying to use graphs for text processing. I want to add graph embeddings to the corresponding text encoder from a syntax graph. For this, each data record has an associated graph which is supposed to be a standalone unit.
For example, for a mock sample data record such as {"A": tensor([1,2,3,4, 5]), "graph": Data(...)}, I want to batch it (which works fine with PyG DataLoader) and run it through a layer that returns a shape like [batch_size, hidden_dim]. Currently, I can batch and propagate batches like in the mock sample, but the shape of the output (trying out GCN) is not as intended, and I cannot figure out how to make embeddings per entry graph without knowledge from other graphs. I could make a forward pass per entry and use AdativeAveragePooling, but I am searching for a solution that can use batching. Does someone have a hint on how can I achieve this behaviour? Thanks in advance for any help you can provide.

**Comment by rusty1s** [comment, marked as answer]
It looks like your model misses out on a global pooling step, right? Have you looked at [these](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#global-pooling-layers) operators? [Here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/mutag_gin.py#L54) is an example on how to use.

**Reply by afonso-sousa** [reply]
Precisely what I needed. Thank you.
